daceml/pytorch/module_codegen.py

```diff
@@ -0,0 +1,287 @@
+        .device(torch::{'kCUDA' if is_cuda(desc.storage) else 'kCPU'})
+        .layout(torch::kStrided));
+    """
+
+
+def initialize_outputs_code(module: 'daceml.pytorch.DaceModule',
+                            output_names: List[str]) -> str:
+    """ Generate the code that initializes the output tensors
+
+        :param module: the module
+        :return: the code
+    """
+    arglist = module.sdfg.arglist()
+    code = ""
+    for name in output_names:
+        code += tensor_init_for_desc(name, arglist[name])
+
+    return code
+
+
+def argument_codegen(module: 'daceml.pytorch.DaceModule',
+                     input_names: List[str],
+                     output_names: List[str]) -> Tuple[str, str, str]:
+    """ Generate the code that grabs the pointers of inputs and outputs.
+
+        :param module: the module
+        :return: the code for initializing the argument, the sdfg arguments in order, and the init call arguments
+    """
+    arglist = module.sdfg.arglist()
+
+    # initialize the inputs and outputs
+    ptr_init_code = "\n    // setup input and output pointers\n    "
+    ptr_init_code += '\n    '.join(
+        f"{arglist[name].dtype.ctype} *{name}_ptr = {name}_.data_ptr<{arglist[name].dtype.ctype}>();"
+        for name in itertools.chain(input_names, output_names))
+    ptr_init_code += "\n    // setup constant arguments\n"
+
+    # initialize all remaining parameters
+    remaining = set(arglist).difference(
+        itertools.chain(input_names, output_names))
+    for name in remaining:
+
+        # remaining args must be constants
+        if name not in module.dace_model.clean_weights:
+            raise ValueError(
+                f"Cannot generate PyTorch module C++ code: SDFG argument {name} is not an input or output"
+                f" of the PyTorch Module, and not a constant.")
+        if arglist[name].total_size > 1000:
+            raise ValueError(
+                f"Cannot generate PyTorch module C++ code: SDFG argument {name} is not an input or output"
+                f" of the PyTorch Module, and is too large.")
+
+        value = module.dace_model.clean_weights[name]
+        ptr_init_code += f"    {constant_initializer_code(name + '_ptr', arglist[name], value)}\n"
+
+    arguments = ", ".join(f"{n}_ptr" for n in arglist)
+    init_arguments = ", ".join(f"{n}_ptr" for n, desc in arglist.items()
+                               if isinstance(desc, data.Scalar))
+
+    return ptr_init_code, arguments, init_arguments
+
+
+def constant_initializer_code(name: str, desc: data.Data, value) -> str:
+    if isinstance(desc, data.Array):
+        iterator = np.nditer(value.cpu().numpy(), order="C")
+        return f"{desc.dtype.ctype} {name}[{desc.total_size}] = {{{', '.join(str(e) for e in iterator)}}};"
+    else:
+        return f"{desc.dtype.ctype} {name} = {str(value.item())};"
+
+
+def code_for_backward_function(module: 'daceml.pytorch.DaceModule',
```

Comment:
do you plan to generate both in the same file?

Reply:
depends on how much code is shared and how long is. At the moment I think so